src/main/java/com/github/havardh/javaflow/Execution.java

```diff
@@ -108,33 +119,46 @@ private void verify(List<Type> types) {
     }
   }
 
+  /**
+   * Resolves all files matching the pattern.
+   *
+   * @param pattern  a file, directory or glob pattern
+   * @return list of files matching the pattern
+   */
+  private List<Path> resolve(List<String> pattern) {
+    return pattern.stream()
+            .map(resolver::resolve)
+            .flatMap(List::stream)
+            .collect(Collectors.toList());
+  }
+
   /**
    * Read each {@code file} into a list of {@code String} with the {@code reader}.
    *
    * @param filenames list of files to read
    * @return list of file content
    */
-  private List<String> read(String[] filenames) {
-    return Stream.of(filenames)
-        .map(reader::read)
-        .filter(Optional::isPresent)
-        .map(Optional::get)
-        .collect(toList());
+  private List<String> read(List<Path> filenames) {
+    return filenames.stream()
+            .map(path -> path.toFile().getPath())
+            .map(reader::read)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(Collectors.toList());
   }
 
   /**
-   * Parse each {@code file} to a list of {@code Type}. The {@code Type} is the
+   * Parse each {@code path} to a list of {@code Type}. The {@code Type} is the
    * internal representation of a model.
    *
-   * @param files list of file contents
+   * @param paths list of file paths
    * @return list of internal types
    */
-  private List<Type> parse(List<String> files) {
-    return files.stream()
+  private List<Type> parse(List<String> paths) {
+    return paths.stream()
```

Comment:
I think this should be named `sources`, right?

Reply:
Yes :)